File: backend/app/models/project.py
import datetime as dt

from sqlalchemy import CheckConstraint, Date, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class Project(Base):
    __tablename__ = "projects"
    __table_args__ = (
        CheckConstraint("end_date >= start_date", name="ck_projects_end_after_start"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    start_date: Mapped[dt.date] = mapped_column(Date, nullable=False)
    end_date: Mapped[dt.date] = mapped_column(Date, nullable=False)

    tasks = relationship("Task", back_populates="project", cascade="all, delete-orphan")

File: frontend/src/types/project.ts
export interface Project {
  id: number;
  name: string;
  description: string | null;
  start_date: string;
  end_date: string;
}

export interface ProjectCreatePayload {
  name: string;
  description: string;
  start_date: string;
  end_date: string;
}
